File: assets/samples/reverse.whois/key.value/ruby/reverse_api_example_post.rb
require 'uri'
require 'net/https'
require 'openssl'

url = URI("https://www.whoisxmlapi.com/reverse-whois-api/search.php")

http = Net::HTTP.new(url.host, url.port)
http.use_ssl = true
http.verify_mode = OpenSSL::SSL::VERIFY_NONE

request = Net::HTTP::Post.new(url)
request["cache-control"] = 'no-cache'
request.body = "{\"terms\": [{\"section\": \"Registrant\", \"attribute\":"\
               + " \"Email\", \"value\": \"support@whoisxmlapi.com\", "\
               + "\"matchType\": \"anywhere\", \"exclude\": \"false\"}],"\
               + "\"recordsCounter\": \"false\", \"outputFormat\": \"json\","\
               + " \"username\": \"Your whois api username\","\
               + " \"password\": \"Your whois api password\""\
               + ", \"rows\": \"10\",  \"searchType\": \"current\"}"

response = http.request(request)
puts response.read_body

File: assets/samples/reverse.whois/key.value/nodejs/reverse_api_example.post.js
var http = require("https");

var options = {
  "method": "POST",
  "hostname": "www.whoisxmlapi.com",
  "port": 443,
  "path": "/reverse-whois-api/search.php",
  "headers": {
    "cache-control": "no-cache"
  }
};

var req = http.request(options, function (res) {

  var chunks = '';
  res.on("data", function (chunk) {chunks += chunk;} );
  res.on("end", function () {
    console.log(chunks);
  });
});
req.write("{\"terms\": [{\"section\": \"Registrant\", \"attribute\":"
  + " \"Email\", \"value\": \"support@whoisxmlapi.com\", \"matchType\":"
  + " \"exact\", \"exclude\": \"false\"}], \"recordsCounter\": \"false\","
  + " \"outputFormat\": \"json\", \"username\": \"your whois api username\", "
  + " \"password\": \"your whois api password\", \"rows\": \"10\", "
  + "\"searchType\": \"current\"}");
req.end();